contrib/src/main/java/com/datatorrent/contrib/geode/GeodeStore.java

```
@@ -0,0 +1,308 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.geode;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.gemstone.gemfire.cache.CacheClosedException;
+import com.gemstone.gemfire.cache.CacheWriterException;
+import com.gemstone.gemfire.cache.EntryNotFoundException;
+import com.gemstone.gemfire.cache.Region;
+import com.gemstone.gemfire.cache.TimeoutException;
+import com.gemstone.gemfire.cache.client.ClientCache;
+import com.gemstone.gemfire.cache.client.ClientCacheFactory;
+import com.gemstone.gemfire.cache.client.ClientRegionShortcut;
+import com.gemstone.gemfire.cache.query.FunctionDomainException;
+import com.gemstone.gemfire.cache.query.NameResolutionException;
+import com.gemstone.gemfire.cache.query.QueryInvocationTargetException;
+import com.gemstone.gemfire.cache.query.SelectResults;
+import com.gemstone.gemfire.cache.query.TypeMismatchException;
+
+import com.datatorrent.lib.db.KeyValueStore;
+
+/**
+ * Provides the implementation of a Geode store.
+ *
+ * 
+ */
+public class GeodeStore implements KeyValueStore, Serializable
+{
+  /**
+   * 
+   */
+  private static final long serialVersionUID = -5076452548893319967L;
+  private static final Logger LOG = LoggerFactory.getLogger(GeodeStore.class);
+  private transient ClientCache clientCache = null;
+  private transient Region<Object, Object> region = null;
+  private String locatorHost;
+  private int locatorPort;
+  private String regionName;
+
+  private ClientCache initClient()
+  {
+    try {
+      clientCache = new ClientCacheFactory().addPoolLocator(getLocatorHost(), getLocatorPort()).create();
+    } catch (CacheClosedException ex) {
+      LOG.info("error initiating client ", ex);
```

Throw exception instead of logging it